From 7c2668d473d24eedef7e67693bd83c35116fe376 Mon Sep 17 00:00:00 2001
From: Mao Minkai <maominkai@wxiat.com>
Date: Tue, 7 Jun 2022 17:41:41 +0800
Subject: [PATCH] sw64: modify tc_sched_clock debugfs file
Patch-mainline: NO, OTHERS
References: openEuler-22.03-LTS
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56OLG

--------------------------------

Use debugfs_create_file() to create debugfs file for tc_sched_clock
so it is protected against file removals. Change its name to
"tc_sched_clock" to stay consistent with what we use in cmdline.

Signed-off-by: Mao Minkai <maominkai@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/kernel/time.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/sw_64/kernel/time.c b/arch/sw_64/kernel/time.c
index 0815d06b03d4..185db832dfb8 100644
--- a/arch/sw_64/kernel/time.c
+++ b/arch/sw_64/kernel/time.c
@@ -237,8 +237,8 @@ static int __init sched_clock_debug_init(void)
 	if (!sw64_debugfs_dir)
 		return -ENODEV;
 
-	sched_clock_status = debugfs_create_file_unsafe("use_tc_as_sched_clock",
-			0666, sw64_debugfs_dir, NULL,
+	sched_clock_status = debugfs_create_file("tc_sched_clock",
+			0644, sw64_debugfs_dir, NULL,
 			&sched_clock_status_fops);
 
 	if (!sched_clock_status)
-- 
2.34.1

